cfme/tests/ansible/test_embedded_ansible_services.py

```diff
@@ -843,3 +843,53 @@ def _revert():
     view = navigate_to(ansible_service, "Details")
     assert view.provisioning.credentials.get_text_of("Vault") == vault_creds.name
     assert view.provisioning.results.get_text_of("Status") == "successful"
+
+
+@pytest.mark.tier(3)
+@pytest.mark.meta(automates=[BZ(1734904)])
+def test_ansible_service_ansible_galaxy_role(
+    appliance,
+    request,
+    ansible_catalog_item,
+    ansible_service,
+    ansible_service_catalog,
+):
+    """Check Role is fetched from Ansible Galaxy by using roles/requirements.yml file
+    from playbook.
+
+    Bugzilla:
+        1734904
+
+    Polarion:
+        assignee: sbulage
+        casecomponent: Ansible
+        initialEstimate: 1/3h
+        tags: ansible_embed
+    """
+    with update(ansible_catalog_item):
+        ansible_catalog_item.provisioning = {
+            "playbook": "ansible_galaxy_role_users.yaml",
+        }
+
+    @request.addfinalizer
+    def _revert():
+        with update(ansible_catalog_item):
+            ansible_catalog_item.provisioning = {
+                "playbook": "dump_all_variables.yml",
+            }
+
+        service = MyService(appliance, ansible_catalog_item.name)
+        if service_request.exists():
```

Comment:
If the test fails on line 888, the `service_request` will be undefined, the `_revert()` will create service and then fail on 882 and fails to delete the service it just created.

Reply:
@jaryn I can see on line 888, if `ansible_service_catalog.order()` fails then it won't create any service_request. Also, in the `_revert()` it is creating instance of `Service` not creating it. Further more every entity is going to check whether `exists` or not.

Reply:
Well the `service_request.exists()` will cause `NameError` when `service_request` is undefined.

Reply:
@jaryn  `exists()` method will return `bool`, hence no `NameError` when service is undefined.

Reply:
@valaparthvi I am agree with you, but the error which you pointed out will fail entire test not the `_revert()`. One more thing if service_request object/variable is not being created then there might be a problem in service catalog `order()` function. I hope this will answer it.